lib/src/mock_collection_reference.dart

```diff
@@ -74,13 +74,15 @@ class MockCollectionReference<T extends Object?> extends MockQuery<T>
         return documentReference.get();
       }).toList();
     }
-    return MockQuerySnapshot<T>(
+    final snapshot = MockQuerySnapshot<T>(
       (await Future.wait(futureDocs))
           .where((snapshot) =>
               _firestore.hasSavedDocument(snapshot.reference.path))
           .toList(),
       options?.source == Source.cache,
     );
+    QuerySnapshotStreamManager().setCacheQuerySnapshot(this, snapshot);
```

Comment:
Seems like you're already updating cacheQuerySnapshot in fireSnapshotUpdate. Why do we need this for each `MockQuery.get`?

Reply:
Yeah actually we can remove query set from `fireSnapshotUpdate`, and only keep in `MockQuery.get`. Because `fireSnapshotUpdate` is only being called when data is being updated/added/deleted, so it means we are not storing initial cache until something is updated, which will result in two consecutive `added` docchange event, first initial from query where cache wasn't set, and second when data is updated and initial cache set but emits added event